src/Project/CommandHandler/ActivateProjectCommandHandler.php

```diff
@@ -92,7 +90,7 @@ public function handle(ActivateProject $activateProject)
         $this->cultureFeedLive->addServiceConsumerAdmin($cultureFeedConsumer->consumerKey, $this->user->id);
 
         // Add uitpas permssion to consumer
-        $this->cultureFeedLive->addUitpasPermission($cultureFeedConsumer, $this->uitpasPermissionGroup);
+        $this->cultureFeedLive->addUitpasPermission($cultureFeedConsumer, $this->permissionGroups['uitpas']);
```

Comment:
@bertramakers very nice suggestion! 
Do you think we can remove this line and just use the code you suggested on line 73.
Something like this:
`$createConsumer->group[] = $this->permissiongGroups[$groupId]`  (which contains the uitpas permission group)?

Reply:
See also this PR: [https://github.com/cultuurnet/culturefeed-php/pull/73](https://github.com/cultuurnet/culturefeed-php/pull/73)

Reply:
Good question. I think I misunderstood the `UiTPAS` permission before. I thought it was also an integration type, but it's actually a permission that needs to be set on UiTPAS itself. If I understand correctly now the permission systems are split up over UiTID and UiTPAS, as described in https://jira.uitdatabank.be/browse/PROJ-118. So we would have to split the permissions up in the `yaml` file as well, for example:

```
permission_groups: 
  # Widgets
  24378: 
    uitid: [3, 24378]
    uitpas: [22678]
  # EntryAPI3
  24380: 
    uitid: [3, 24380, 24640]
    uitpas: [22678]
```

Then in the code above we could do:
```
$integrationType = $createProject->getIntegrationType();
$createConsumer->group[] = $this->permissionGroups[$integrationType]['uitid'];
```

And for the uitpas permission(s) here, we could do:
```
$this->cultureFeedLive->addUitpasPermission($cultureFeedConsumer, $this->permissionGroups[$integrationType]['uitpas']);
```

At least I think the UiTPAS API allows an array of permissions...
The documentation says this for the `group` parameter sent with`addUitpasPermission()`: 

> Alle UiTPAS groepen van deze consumer. De parameter kan herhaald worden om meerdere groepen aan de consumer toe te kennen.Bij elke request moet altijd de volledige set van groepen opnieuw meegeven worden. Niet meegegeven groepen, worden verwijderd van de consumer.

But how/if culturefeed-php handles that under the hood is sometimes a mystery to me as well. It's _very_ old code. 😕

Reply:
Worst case we make the `uitpas` part in the `yaml` config just a single integer for now.